File: lib/models/meditation_models.dart
enum MeditationCategory {
  sleep,
  focus,
  anxiety,
  stress,
  mindfulness,
  beginner
}

extension MeditationCategoryExtension on MeditationCategory {
  String get displayName {
    switch (this) {
      case MeditationCategory.sleep:
        return 'Sueño';
      case MeditationCategory.focus:
        return 'Concentración';
      case MeditationCategory.anxiety:
        return 'Ansiedad';
      case MeditationCategory.stress:
        return 'Estrés';
      case MeditationCategory.mindfulness:
        return 'Mindfulness';
      case MeditationCategory.beginner:
        return 'Principiantes';
    }
  }
}

class MeditationSession {
  final String id;
  final String title;
  final String description;
  final int duration; // en segundos
  final MeditationCategory category;
  final String audioUrl;
  final String imageUrl;
  final bool isPremium;

  MeditationSession({
    required this.id,
    required this.title,
    required this.description,
    required this.duration,
    required this.category,
    required this.audioUrl,
    required this.imageUrl,
    this.isPremium = false,
  });
}

class MeditationProgress {
  final String sessionId;
  final int completedTime;
  final int totalTime;
  final int lastPlayedPosition;

  MeditationProgress({
    required this.sessionId,
    required this.completedTime,
    required this.totalTime,
    required this.lastPlayedPosition,
  });
} 

File: app/src/main/java/com/mindcare/mindcare/meditation/models/MeditationModels.kt
package com.mindcare.mindcare.meditation.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "meditation_sessions")
data class MeditationSession(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val meditationId: Long,
    val startTime: Date,
    val duration: Int, // en minutos
    val completed: Boolean,
    val focusScore: Float?, // 0.0 a 1.0
    val notes: String?
)

data class Meditation(
    val id: Long,
    val title: String,
    val description: String,
    val category: MeditationCategory,
    val duration: Int, // en minutos
    val audioUrl: String,
    val imageUrl: String,
    val difficulty: MeditationDifficulty,
    val tags: List<String>,
    val guidedSteps: List<GuidedStep>
)

data class GuidedStep(
    val timestamp: Int, // segundos desde el inicio
    val instruction: String,
    val duration: Int // duración de la instrucción en segundos
)

enum class MeditationCategory {
    MINDFULNESS,
    BREATHING,
    BODY_SCAN,
    LOVING_KINDNESS,
    STRESS_RELIEF,
    SLEEP,
    FOCUS,
    ANXIETY_RELIEF
}

enum class MeditationDifficulty {
    BEGINNER,
    INTERMEDIATE,
    ADVANCED
}

data class MeditationProgress(
    val totalSessions: Int,
    val totalMinutes: Int,
    val averageFocusScore: Float,
    val longestStreak: Int,
    val currentStreak: Int,
    val favoriteCategory: MeditationCategory?,
    val lastSession: Date?
) 